src/rdkafka.h

```diff
@@ -1727,7 +1727,73 @@ void rd_kafka_conf_set_stats_cb(rd_kafka_conf_t *conf,
 						  size_t json_len,
 						  void *opaque));
 
+/**
+ * @brief Set SASL/OAUTHBEARER token refresh callback in provided conf object.
+ *
+ * The SASL/OAUTHBEARER token refresh callback is triggered via rd_kafka_poll()
+ * whenever OAUTHBEARER is the SASL mechanism and a token needs to be retrieved,
+ * typically based on the configuration defined in \c sasl.oauthbearer.config.
+ * Function arguments:
+ *   - \p rk - Kafka handle
+ *   - \p opaque - Application-provided opaque.
+ * 
+ * The callback must invoke \c rd_kafka_oauthbearer_token_refresh_success()
+ * or \c rd_kafka_oauthbearer_token_refresh_failure() to indicate success
+ * or failure, respectively.
+ * 
+ * The refresh operation is eventable and may be received via
+ * rd_kafka_queue_poll() with an event type of
+ * RD_KAFKA_EVENT_OAUTHBEARER_TOKEN_REFRESH.
+ * 
+ * Note that the application must call either rd_kafka_poll() or
+ * rd_kafka_queue_poll() once before any broker connection is attempted
+ * in order to retrieve an initial token.
+ */
+RD_EXPORT
+void rd_kafka_conf_set_oauthbearer_token_refresh_cb(rd_kafka_conf_t *conf,
+                void (*oauthbearer_token_refresh_cb) (rd_kafka_t *rk,
+                void *opaque));
+
+/**
+ * @brief SASL/OAUTHBEARER token refresh success indicator.
+ *
+ * The SASL/OAUTHBEARER token refresh callback or event handler must invoke
+ * this method uopn success. The caller remains responsible for the token_value
+ * and md_principal_name memory as the contents are copied.  The md_start_time_ms
+ * value should be 0 if it is not known; all other arguments are mandatory.
+ * 
```

Comment:
👍 Added comment stating "`The md_lifetime_ms value is when the token expires, in terms of the number of milliseconds since the epoch.`" . The start time is now removed as per another review comment.